scripts/o3de/o3de/github_utils.py

```diff
@@ -86,6 +87,81 @@ def clone_from_git(self, uri: ParseResult, download_path: pathlib.Path, force_ov
                 return 1
 
         return proc.returncode
+    
+    def upload_release_to_github(self, repo_uri: ParseResult, zip_path: pathlib.Path, archive_filename: str, upload_git_release_tag: str):
```

Comment:
lets have the error paths return 1 on failure so we can use the error code to detect failure if running this command from a shell script.